### PR TITLE
files: add files in the document detailed view

### PR DESCRIPTION
**Pull request by jma**
* Adds the file list in the document detailed view using primeng.
* Adds "search in fulltext" button in the professionnal and public view.
* Changes the html page title for the document holding files view.

### CODE REVIEW

File: projects/shared/src/lib/component/documents/files/files.component.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!--
+  RERO ILS UI
+  Copyright (C) 2021 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.html:3]: 2024 ?

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019-2023 RERO
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:3]: 2024

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  templateUrl: "./files.component.html",
+  styleUrl: "./files.component.scss",
+})
+export class FilesComponent implements OnInit {
+
+  // input document pid
+  @Input() documentPid: string;
+
+  // list of files
+  files : Array<File> = [];
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:47]: `File[]`

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  ];
+
+  // primeng configuration service
+  private ngConfigService = inject(PrimeNGConfig);
+  // http service
+  private httpService = inject(HttpClient);
+  // ng-core record service
+  private recordService =  inject(RecordService);
+  // ng-core api service
+  private apiService = inject(ApiService);
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:75]: Nice 👍🏻

File: projects/shared/src/lib/component/documents/files/files.component.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    this.ngConfigService.translation.aria.slideNumber = "{slideNumber}";
+  }
+
+
+  /** OnInit hook */
+  ngOnInit() {
+    // default thumbnail
+    // TODO: change this according to the PO
+    const thumbnail =
+      "/static/thumbnails/icon_docmaintype_book.svg";
```
Comment (Garfield-fr) [inline projects/shared/src/lib/component/documents/files/files.component.ts:90]: Why hardcoded the path ?
Suggestion: Make the path to environment ?

**Review by Garfield-fr — APPROVED** [review]

File: projects/shared/src/lib/component/documents/files/files.component.ts (hunk 5 above)
Comment (jma) [reply projects/shared/src/lib/component/documents/files/files.component.ts:90]: this is solved on the next pull requests.

File: projects/shared/src/lib/component/documents/files/files.component.ts (hunk 3 above)
Comment (jma) [reply projects/shared/src/lib/component/documents/files/files.component.ts:47]: Both are similar. Which one is better? Any recommandation?

Comment (Garfield-fr) [reply projects/shared/src/lib/component/documents/files/files.component.ts]: Angular code uses `File[]`. This is the syntax we had chosen to pursue the devs 😉

**Pull request merged**